packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -346,6 +346,16 @@ const bar = foo.a as string | undefined | bigint;
       `,
       parserOptions: optionsWithExactOptionalPropertyTypes,
     },
+    {
+      code: `
+if (Math.random()) {
+  {
+    var x = 1;
+  }
+}
+x!;
+      `,
+    },
```

Comment:
[Testing] Could you also add an `invalid` case that shows a `var` inside an `if`? Generally when testing rules, if there's a case that's _valid_, it's common to want to have a corresponding _invalid_ case.

Reply:
The current implementation ignores any child scoped `var` to avoid false positives, and I haven't figured out a corresponding invalidation test yet.🤔

Reply:
Right, great 😄